SingleSource/Regression/C/gcc-c-torture/execute/comp-goto-1.c

```diff
@@ -9,7 +9,11 @@ typedef signed int sint32;
 
 typedef uint32 reg_t;
 
+#ifdef __CHERI_PURE_CAPABILITY__
+typedef char *host_addr_t;
```

Comment:
Can we just use uintptr_t here and for rigged_paddr? not sure if that's enough to fix the program but it seems simpler.

Reply:
The problem is that the program constructs a substantially out-of-bounds pointer and assigns it to  `rigged_paddr`.  It's too far out-of-bounds and is tagged as undereferencable.
(At line 168, `a_page` points to an 8192 byte region returned from `malloc`, and the value `0x123000` is subtracted from it.)